components/discardable_memory/client/client_discardable_shared_memory_manager.cc

```diff
@@ -361,8 +368,28 @@ ClientDiscardableSharedMemoryManager::AllocateLockedDiscardableSharedMemory(
                "size", size, "id", id);
 #if defined(CASTANETS)
   auto memory = base::MakeUnique<base::DiscardableSharedMemory>();
-  if (!memory->CreateAndMap(size))
-    base::TerminateBecauseOutOfMemory(size);
+  if (base::Castanets::IsEnabled()) {
+    memory = base::MakeUnique<base::DiscardableSharedMemory>();
```

Comment:
This line is not necessary.

Reply:
IMO, it is needed, AFAIK there was a crash due to this.

Reply:
'memory' has to be initialized two times?

#if defined(CASTANETS)
  auto memory = base::MakeUnique<base::DiscardableSharedMemory>();
  if (base::Castanets::IsEnabled()) {                                                                                                                                                                       
    memory = base::MakeUnique<base::DiscardableSharedMemory>();

Reply:
Yes, memory is defined as auto => to know type of variable we need this definition

Reply:
#if defined(CASTANETS)
  auto memory = **base::MakeUniquebase::DiscardableSharedMemory();** <-- Define & init
  if (base::Castanets::IsEnabled()) {
    memory = **base::MakeUniquebase::DiscardableSharedMemory();**  <--  Is necessary?

Reply:
got your point, multiple initiations are not needed, will handle this, thanks!